kratos/geometries/triangle_3d_3.h

```diff
@@ -879,7 +875,66 @@ template<class TPointType> class Triangle3D3
         const CoordinatesArrayType& rPoint
         ) const override
     {
-        return PointLocalCoordinatesImplementation(rResult, rPoint);
+        // Initialize
+        rResult = ZeroVector(3);
+
+        // Tangent vectors
+        const array_1d<double, 3> tangent_xi  = this->GetPoint(1) - this->GetPoint(0);
+        const array_1d<double, 3> tangent_eta = this->GetPoint(2) - this->GetPoint(0);
+
+        // We compute the normal to check the normal distances between the point and the triangles, so we can discard that is on the triangle
+        array_1d<double, 3> normal;
+        MathUtils<double>::UnitCrossProduct(normal, tangent_xi, tangent_eta);
+
+        // The center of the geometry
+        const auto& r_center = this->Center();
+
+        // We compute the distance, if it is not in the pane we
+        CoordinatesArrayType point_projected = rPoint;
+        const array_1d<double,3> vector_points = rPoint - r_center.Coordinates();
+        const double distance = inner_prod(vector_points, normal);
+        if (std::abs(distance) > std::numeric_limits<double>::epsilon()) { // Not in the plane, projecting
+            noalias(point_projected) = rPoint - normal * distance;
+        }
+
+        // Computation of the rotation matrix
+        BoundedMatrix<double, 3, 3> rotation_matrix = ZeroMatrix(3, 3);
+        for (IndexType i = 0; i < 3; ++i) {
+            rotation_matrix(0, i) = tangent_xi[i];
+            rotation_matrix(1, i) = tangent_eta[i];
+        }

```

Comment:
Don't we need the third row to be the normal[i] here?

Reply:
This is the algorithm I use in the mortar segmentation, i think is not needed:

https://github.com/KratosMultiphysics/Kratos/blob/2ef11c5c6baebc205d4e25606254a3059a14d18b/kratos/utilities/mortar_utilities.h#L185

Reply:
Ok, you know your algorithms… Just saying that this is not a rotation matrix then :P

Reply:
True, projection to base matrix?